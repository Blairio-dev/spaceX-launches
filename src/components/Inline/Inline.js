import React from "react";
import PropTypes from "prop-types";
import styled from "@emotion/styled";
import { breakpoints } from "../../assets/tokens";

const StyledWrapper = styled("div")`
  align-items: ${(props) => props.alignItems};
  display: flex;
  justify-content: ${(props) => props.justification};
  width: 100%;

  @media (max-width: ${breakpoints.lg}) {
    flex-flow: wrap-reverse;
  }
`;

const getVerticalAlignment = (verticalAlignment) => {
  switch (verticalAlignment) {
    case "baseline":
      return "baseline";
    case "bottom":
      return "flex-end";
    case "center":
      return "center";
    case "top":
      return "flex-start";
    default:
      return "stretch";
  }
};

const BaseInline = ({ justification, nodes, verticalAlignment }) => (
  <StyledWrapper
    alignItems={getVerticalAlignment(verticalAlignment)}
    justification={justification}
  >
    {nodes.map((component) => component)}
  </StyledWrapper>
);

const baseProps = {
  /** An array of components. */
  nodes: PropTypes.arrayOf(PropTypes.node).isRequired,
  /** Vertical alignment of components within flex containter. */
  verticalAlignment: PropTypes.oneOf([
    "baseline",
    "bottom",
    "center",
    "stretch",
    "top",
  ]),
};

BaseInline.propTypes = {
  ...baseProps,
  /** Jusification of components within flex containter. */
  justification: PropTypes.string,
};

BaseInline.defaultProps = {
  verticalAlignment: "stretch",
};

const Justified = ({ nodes, verticalAlignment }) => (
  <BaseInline
    justification="space-between"
    nodes={nodes}
    verticalAlignment={verticalAlignment}
  />
);

Justified.propTypes = {
  ...baseProps,
};

const RightAligned = ({ nodes, verticalAlignment }) => (
  <BaseInline
    justification="flex-end"
    nodes={nodes}
    verticalAlignment={verticalAlignment}
  />
);

RightAligned.propTypes = {
  ...baseProps,
};

const Inline = {
  Justified,
  RightAligned,
};

export { Inline };
