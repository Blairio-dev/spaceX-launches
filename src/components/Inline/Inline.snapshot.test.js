import React from "react";
import renderer from "react-test-renderer";
import { Inline } from "./Inline";

const DEFAULT_PROPS = {
  nodes: [<div key="123">123</div>, <div key="345">345</div>],
};

const FULLEST_PROPS = {
  ...DEFAULT_PROPS,
  verticalAlignment: "center",
};

describe("Inline Snapshots", () => {
  describe(".Justified", () => {
    it("Renders with default props", () => {
      const tree = renderer
        .create(<Inline.Justified {...DEFAULT_PROPS} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("Renders with fullest props", () => {
      const tree = renderer
        .create(<Inline.Justified {...FULLEST_PROPS} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe(".RightAligned", () => {
    it("Renders with default props", () => {
      const tree = renderer
        .create(<Inline.RightAligned {...DEFAULT_PROPS} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    it("Renders with fullest props", () => {
      const tree = renderer
        .create(<Inline.RightAligned {...FULLEST_PROPS} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
