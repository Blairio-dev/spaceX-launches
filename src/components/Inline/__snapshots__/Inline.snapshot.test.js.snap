// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inline Snapshots .Justified Renders with default props 1`] = `
<div
  className="css-hheasg"
>
  <div>
    123
  </div>
  <div>
    345
  </div>
</div>
`;

exports[`Inline Snapshots .Justified Renders with fullest props 1`] = `
<div
  className="css-1ff6szg"
>
  <div>
    123
  </div>
  <div>
    345
  </div>
</div>
`;

exports[`Inline Snapshots .RightAligned Renders with default props 1`] = `
<div
  className="css-17w8ll9"
>
  <div>
    123
  </div>
  <div>
    345
  </div>
</div>
`;

exports[`Inline Snapshots .RightAligned Renders with fullest props 1`] = `
<div
  className="css-ck5tno"
>
  <div>
    123
  </div>
  <div>
    345
  </div>
</div>
`;
